packages/manager/src/features/Users/userPermissionsReducer.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  GlobalGrantTypes,
+  Grant,
+  GrantLevel,
+  GrantType,
+  Grants,
+} from '@linode/api-v4/lib/account';
+
+export type PermSelectValue = 'null' | 'read_only' | 'read_write' | null;
+
+export interface UserGrantsState {
+  grants: Grants[];
+  originalGrants: Grants[];
+}
+
+export const initialState: UserGrantsState = {
+  grants: [],
+  originalGrants: [],
+};
+
+export type Action =
+  | {
+      changedGlobalPerm: string;
+      type: 'GLOBAL_PERM_CHANGED';
+      value: boolean;
+    }
+  | {
+      entity: GrantType;
+      entityPerms: GrantType[];
+      type: 'ALL_SPECIFIC_PERMS_CHANGED';
+    }
+  | {
+      entity: string;
+      idx: number;
+      type: 'SPECIFIC_PERM_CHANGED';
+      value: GrantLevel;
+    }
+  | {
+      entity: string;
+      type: 'SPECIFIC_PERMS_SAVED';
+      value: Grant;
+    }
+  | {
+      grants: any;
+      type: 'GRANTS_SET';
+    }
+  | {
+      savedGlobalPerms: Record<GlobalGrantTypes, GrantLevel | boolean>;
+      type: 'GLOBAL_PERMS_SAVED';
+    }
+  | { billingAccess: PermSelectValue; type: 'BILLING_ACCESS_CHANGED' }
+  | { originalGrants: any; type: 'ORIGINAL_GRANTS_SET' }
```

Comment:
Type should be` Grants[]`
```suggestion
  | { originalGrants: Grants[]; type: 'ORIGINAL_GRANTS_SET' }
```